test_run.lua

```diff
@@ -65,9 +65,10 @@ local function wait_lsn(self, waiter, master)
     local sid = self:get_server_id(master)
     local lsn = self:get_lsn(master, sid)
 
-    while self:get_lsn(waiter, sid) < lsn do
+    repeat
         fiber.sleep(0.001)
-    end
+        local waiter_lsn = self:get_lsn(waiter, sid)
+    until waiter_lsn ~= nil and waiter_lsn >= lsn
```

Comment:
Maybe `waiter_lsn < lsn`?
AFAIU @Totktonada suggests to continue waiting in case of an empty `vclock`.

Reply:
Seems you missed that I've used until in place of while.

Reply:
D'oh! I confused the behavior of `do {...} while (condition)` with the `repeat ... until condition`.